Modules/StudyProgramme/classes/class.ilObjStudyProgrammeMembersGUI.php

```diff
@@ -950,18 +950,16 @@ public function mayManageMembers() : bool
         );
     }
 
-    public function getLocalMembers(): array
+    public function getLocalMembers() : array
     {
         return $this->object->getMembers();
     }
 
-    public function isOperationAllowedForUser(int $usr_id, string $operation): bool
+    public function isOperationAllowedForUser(int $usr_id, string $operation) : bool
     {
-        $ret = $this->position_based_access->isUserAccessibleForOperationAtPrg(
-            $usr_id,
-            $this->object,
-            $operation
-        );
+        $ret = $this->position_based_access->isUserAccessibleForOperationAtPrg($usr_id, $this->object, $operation)
+            || $this->access->checkAccess("manage_members", "", $this->object->getRefId());
```

Comment:
Let's switch the operators of the `||` and do the check via the RBAC first. `||` has short circuit behaviour in PHP. That means, if for `a || b` the `a` already evaluates to true, `b` is never evaluated. Since RBAC checks are simpler, this will save time here. Also: Please just return the result instead directly, instead of assigning it to `$ret` before returning this then afterwards,